runtime/ts/planner.ts

```diff
@@ -171,14 +172,14 @@ export class Planner {
         // before returning.
         relevance.newArc.stop();
 
-        results.push({
-          plan,
-          rank,
-          description: relevance.newArc.description,
-          descriptionText: description, // TODO(mmandlis): exclude the text description from returned results.
-          hash,
-          groupIndex
-        });
+        const suggestion = new Suggestion(plan, this._arc);
+        suggestion.rank = rank;
```

Comment:
likewise, rank should be part of the constructor.

If rank is only set from the planner, and never modified, then it can be readonly too, which makes the Suggestion class more readable as a whole. Likewise for hash.